### PR TITLE
Fd 1287 factable updates march to factom

### PR DESCRIPTION
**Pull request by factablesolutions**
Test to see if this pull request is correctly run through regressions. Contains my december updates rebased, and new changes.

### CODE REVIEW

**Comment by stackdump** [comment]
The test that's failing is flaky `TestAuthenticatedUnauthorizedRequest` - We'll work on getting this merged soon

File: common/identity/identityManagerEntryBlock.go [hunk 1]
```diff
@@ -220,6 +221,8 @@ func (im *IdentityManager) ProcessIdentityEntryWithABlockUpdate(entry interfaces
 	return change, nil
 }
 
+// ApplyIdentityChainStructure updates or adds a new identity with the 4 secret keys and the block height creation associated with the chain id.
+// Will also register then identity if it hasn't been registered yet
```
Comment (WhoSoup) [inline common/identity/identityManagerEntryBlock.go]: *the

File: common/identity/identityManagerEntryBlock.go [hunk 2]
```diff
@@ -245,7 +248,7 @@ func (im *IdentityManager) ApplyIdentityChainStructure(ic *IdentityChainStructur
 	return false, nil
 }
 
-//		Returns
+// ApplyNewBitcoinKeyStructure returns
```
Comment (WhoSoup) [inline common/identity/identityManagerEntryBlock.go]: returns what?

File: common/identity/util.go [hunk 3]
```diff
@@ -33,11 +33,11 @@ func CheckTimestamp(time []byte, timeEntered int64) bool {
 	}
 	if res <= TWELVE_HOURS_S {
 		return true
-	} else {
-		return false
 	}
+	return false
```
Comment (WhoSoup) [inline common/identity/util.go]: You can just do `return res <= TWELVE_HOURS_S`

File: common/identity/util.go [hunk 4]
```diff
@@ -49,6 +49,7 @@ func BubbleSortUint32(arr []uint32) []uint32 {
 	return arr
 }
 
+// statusFedOrAudit returns true if the input status is a federated or audit server
```
Comment (WhoSoup) [inline common/identity/util.go]: *statusIsFedOrAudit

File: common/identityEntries/identityKeys.go [hunk 5]
```diff
@@ -5,19 +5,20 @@ package identityEntries
...
 //- Add conversion to human-readable private / public key
 
+// Constants for the prefix part of the identity secret and public keys
 const (
-	IdentityPrivateKeyPrefix1 = "4db6c9"
-	IdentityPrivateKeyPrefix2 = "4db6e7"
-	IdentityPrivateKeyPrefix3 = "4db705"
-	IdentityPrivateKeyPrefix4 = "4db723"
+	IdentityPrivateKeyPrefix1 = "4db6c9" // prefix of "sk1"
+	IdentityPrivateKeyPrefix2 = "4db6e7" // prifix of "sk2"
```
Comment (WhoSoup) [inline common/identityEntries/identityKeys.go]: *prefix

File: common/identityEntries/newCoinbaseAddress_test.go [hunk 6]
```diff
@@ -11,6 +11,7 @@ import (
 	. "github.com/FactomProject/factomd/common/identityEntries"
 )
 
+// TestNewCoinbaseAddressStruct checks a hardcoded external ID can be set into the coinbase address structure and obtain the correct chain ID
```
Comment (WhoSoup) [inline common/identityEntries/newCoinbaseAddress_test.go]: *checks if?

File: common/interfaces/addresses.go [hunk 7]
```diff
@@ -4,27 +4,30 @@
...
-	GetAmount() uint64
-	SetAmount(uint64)
-	GetAddress() IAddress
-	SetAddress(IAddress)
-	IsSameAs(ITransAddress) bool
+	GetAmount() uint64           // Get the amount associated with this transaction
+	SetAmount(uint64)            // Set the amount associated with this transaction
+	GetAddress() IAddress        // Return the address associated with this transaction
+	SetAddress(IAddress)         // Set the address associated with this transaction
+	IsSameAs(ITransAddress) bool // Compares two ITransAddress's
```
Comment (WhoSoup) [inline common/interfaces/addresses.go]: *ITransAddresses?

File: common/interfaces/addresses.go [hunk 8]
```diff
@@ -4,27 +4,30 @@
...
+	CustomMarshalTextInput() ([]byte, error)    // Convert this object into text bytes
+	CustomMarshalTextOutput() ([]byte, error)   // Convert this object into text bytes
+	CustomMarshalTextECOutput() ([]byte, error) // Convert this object into text bytes
 
-	StringInput() string
-	StringOutput() string
-	StringECOutput() string
+	StringInput() string    // Convert this object to a string
+	StringOutput() string   // Convert this object to a string
+	StringECOutput() string // Conver tthis object to a string
```
Comment (WhoSoup) [inline common/interfaces/addresses.go]: *Convert this

File: common/interfaces/binary.go [hunk 9]
```diff
@@ -8,14 +8,16 @@ import (
...
 
 	UnmarshalBinaryData([]byte) ([]byte, error)
 }
 
+// BinaryMarshallableAndCopyable represents an object which is binary marshallable/unmarshallable and copyable
 type BinaryMarshallableAndCopyable interface {
-	BinaryMarshallable
-	New() BinaryMarshallableAndCopyable
+	BinaryMarshallable                  // All of the above
+	New() BinaryMarshallableAndCopyable // Plust the copy piece
```
Comment (WhoSoup) [inline common/interfaces/binary.go]: *Plus

File: common/messages/ack.go [hunk 10]
```diff
@@ -75,14 +80,17 @@ func (m *Ack) GetMsgHash() (rval interfaces.IHash) {
 	return m.MsgHash
 }
 
+// Type returns this message type ACK_MSG
 func (m *Ack) Type() byte {
 	return constants.ACK_MSG
 }
 
+// GetTimestamp returns this objects timestamp
```
Comment (WhoSoup) [inline common/messages/ack.go]: *object's

**Pull request merged**